backend/src/user/user.service.ts

```diff
@@ -140,14 +145,32 @@ export class UserService {
     return leaderboard;
   }
 
-  async addGames(payload, user) {
-    await this.prisma.user.update({
+  async getUserByUsername(dto: GetUserByUsernameDTO): Promise<SecureUser> {
+    return await this.prisma.user.findUnique({
       where: {
-        email: user.email,
+        userName: dto.userName,
       },
-      data: {
-        gamesWon: payload.won,
-        gamesLost: payload.lost,
+      select: {
+        userName: true,
+        avatar: true,
+        status: true,
+        gamesWon: true,
+        gamesLost: true,
+      },
+    });
+  }
+
+  async getUserByEmail(dto: GetUserByEmailDTO): Promise<SecureUser> {
+    return await this.prisma.user.findUnique({
+      where: {
+        email: dto.email,
+      },
+      select: {
+        userName: true,
+        avatar: true,
+        status: true,
+        gamesWon: true,
+        gamesLost: true,
       },
     });
```

Comment:
Probably need to wrap all prisma calls inside `try...catch` to avoid the error from stopping the server.

Reply:
I checked this, try/catch seems to be necessary for socket.io events, but not for HTTP requests, because in this case NestJS middleware handles prisma errors. I didn't use try/catch for prisma calls in the auth module either.